packages/server/src/project/__tests__/project.service.spec.ts

```diff
@@ -0,0 +1,406 @@
+import { ProjectService } from '../project.service';
+import { ConfigService } from '@nestjs/config';
+import * as sinon from 'sinon';
+import { PrismaService } from '../../prisma/prisma.service';
+import { UserService } from '../../user/user.service';
+import { Prisma, Project } from '@prisma/client';
+import { NotFoundException } from '@nestjs/common';
+import { ConfigurableProjectSettings, ProjectAuthMethodsInput, ProjectSettingsInput } from '../dto/project.dto';
+
+const sandbox = sinon.createSandbox();
+
+const MOCK_PROJECT = {
+  id: '1',
+  name: 'sail'
+};
+
+const MOCK_USER = {
+  id: '1',
+  name: 'chris'
+};
+
+const MOCK_SETTINGS = {
+  displayProjectName: true,
+  allowSignup: true
+};
+
+const MOCK_AUTH_METHODS = {
+  googleAuth: true,
+  emailAuth: true
+};
+
+const MOCK_UPDATED_PROJECT = {
+  id: '1',
+  name: 'Updated Project'
+};
+
+describe('ProjectService', () => {
+  let projectService: ProjectService;
+  let mockPrismaService: sinon.SinonStubbedInstance<PrismaService>;
+  let mockUserService: sinon.SinonStubbedInstance<UserService>;
+  let mockConfigService: sinon.SinonStubbedInstance<ConfigService>;
+
+  beforeEach(async () => {
+    mockPrismaService = sandbox.createStubInstance(PrismaService);
+    mockUserService = sandbox.createStubInstance(UserService);
+    mockConfigService = sandbox.createStubInstance(ConfigService);
+    mockConfigService.getOrThrow.returns('');
+
+    mockPrismaService.project = {
+      create: sandbox.stub().returns(MOCK_PROJECT),
+      findMany: sandbox.stub().resolves([MOCK_PROJECT]),
+      findFirstOrThrow: sandbox.stub().returns(MOCK_PROJECT),
+      update: sandbox.stub().returns(MOCK_PROJECT),
+      count: sandbox.stub().returns(1)
```

Comment:
In each on of your tests, you arrange the stubs so you don't need todo that here as well.
```suggestion
      create: sandbox.stub(),
      findMany: sandbox.stub(),
      findFirstOrThrow: sandbox.stub(),
      update: sandbox.stub(),
      count: sandbox.stub()
```